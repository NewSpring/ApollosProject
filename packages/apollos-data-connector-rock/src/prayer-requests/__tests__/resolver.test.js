import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import ApollosConfig from '@apollosproject/config';

import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import {
  peopleSchema,
  mediaSchema,
  authSchema,
} from '@apollosproject/data-schema';

// we import the root-level schema and resolver so we test the entire integration:
import * as Person from '../index';
import authMock from '../../authMock';

const Auth = { schema: authSchema, dataSource: authMock };
const { getContext, getSchema } = createTestHelpers({ Person, Auth });

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://beta-rock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://beta-rock.newspring.cc/GetImage.ashx',
  },
});

describe('PrayerRequest', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([peopleSchema, mediaSchema]);
    context = getContext();
  });

  it("updates a user's attributes, if there is a current user", async () => {
    const query = `
      query {
        getPrayerRequests {
          id
          text
        }
      }
    `;

    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
});
