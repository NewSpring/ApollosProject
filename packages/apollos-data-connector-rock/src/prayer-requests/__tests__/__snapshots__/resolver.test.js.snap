// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrayerRequest updates a user's attributes, if there is a current user 1`] = `
Object {
  "data": Object {
    "getPrayerRequests": null,
  },
  "errors": Array [
    [GraphQLError: Cannot read property 'getPrayerRequests' of undefined],
  ],
}
`;
